fs/attr.c

```diff
@@ -98,8 +144,27 @@ int setattr_prepare(struct dentry *dentry, struct iattr *attr)
 
 	/* Check for setting the inode time. */
 	if (ia_valid & (ATTR_MTIME_SET | ATTR_ATIME_SET | ATTR_TIMES_SET)) {
+#if CONFIG_TRUENAS
+		/*
+		 * Check for ACE4_WRITE_ATTRIBUTES. RFC 5661 Section 6.2.1.3.1
+		 * Permission to change times associated with file or directory
+		 * to an arbitrary value.
+		 */
```

Comment:
I'll need to double-check this one. IIRC there is a nuance where WRITE_DATA / WRITE_ATTRIBUTES allows setting to current time, but not arbitrary time.